rpcs3/Emu/Cell/PPUModule.cpp

```diff
@@ -1544,3 +1546,284 @@ void ppu_load_exec(const ppu_exec_object& elf)
+			}
+			break;
+		}
+
+		case 0x60000002: // LOOS+2 seems to be 0x0 in size for overlay elfs, at least in known cases
+		{
+			if (prog.p_filesz)
+			{
+				struct ppu_proc_prx_param_t
+				{
+					be_t<u32> size;
+					be_t<u32> magic;
+					be_t<u32> version;
+					be_t<u32> unk0;
+					be_t<u32> libent_start;
+					be_t<u32> libent_end;
+					be_t<u32> libstub_start;
+					be_t<u32> libstub_end;
+					be_t<u16> ver;
+					be_t<u16> unk1;
+					be_t<u32> unk2;
+				};
+
+				const auto& proc_prx_param = vm::_ref<const ppu_proc_prx_param_t>(vm::cast(prog.p_vaddr, HERE));
+
+				LOG_NOTICE(LOADER, "* libent_start = *0x%x", proc_prx_param.libent_start);
+				LOG_NOTICE(LOADER, "* libstub_start = *0x%x", proc_prx_param.libstub_start);
+				LOG_NOTICE(LOADER, "* unk0 = 0x%x", proc_prx_param.unk0);
+				LOG_NOTICE(LOADER, "* unk2 = 0x%x", proc_prx_param.unk2);
+
+				if (proc_prx_param.magic != 0x1b434cec)
+				{
+					fmt::throw_exception("Bad magic! (0x%x)", proc_prx_param.magic);
+				}
+
+				ppu_load_exports(link, proc_prx_param.libent_start, proc_prx_param.libent_end);
+				ppu_load_imports(_main->relocs, link, proc_prx_param.libstub_start, proc_prx_param.libstub_end);
+			}
+			break;
+		}
+		default:
+		{
+			LOG_ERROR(LOADER, "Unknown phdr type (0x%08x)", p_type);
+		}
+		}
+	}
+
+	// Initialize process
+	std::vector<std::shared_ptr<lv2_prx>> loaded_modules;
+
+	// Get LLE module list
+	std::set<std::string> load_libs;
+
+	// seems those selfs have no libraries dependencies, imports or exports, but leaving some part of it for now
+
+	if (!load_libs.empty())
+	{
+		const std::string lle_dir = vfs::get("/dev_flash/sys/external/");
+
+		if (!fs::is_dir(lle_dir))
+		{
+			LOG_ERROR(GENERAL, "/dev_flash/sys/external/ directory does not exist!"
+				"\nYou should install the PS3 Firmware (Menu: File -> Install Firmware)."
+				"\nVisit https://rpcs3.net/ for Quickstart Guide and more information.");
+		}
+
+		for (const auto& name : load_libs)
+		{
+			const ppu_prx_object obj = decrypt_self(fs::file(lle_dir + name));
+
+			if (obj == elf_error::ok)
+			{
+				LOG_WARNING(LOADER, "Loading library: %s", name);
+
+				auto prx = ppu_load_prx(obj, lle_dir + name);
+
+				if (prx->funcs.empty())
+				{
+					LOG_FATAL(LOADER, "Module %s has no functions!", name);
+				}
+				else
+				{
+					// TODO: fix arguments
+					prx->validate(prx->funcs[0].addr);
+				}
+
+				loaded_modules.emplace_back(std::move(prx));
+			}
+			else
+			{
+				fmt::throw_exception("Failed to load /dev_flash/sys/external/%s: %s", name, obj.get_error());
+			}
+		}
+	}
+
+	// Set path (TODO)
+	_main->path = vfs::get(path);
+	_main->name = path.substr(path.find_last_of('/') + 1);
+
+	// Analyse executable (TODO)
+	_main->analyse(0, static_cast<u32>(elf.header.e_entry));
+
+	// Validate analyser results (not required)
+	_main->validate(0);
+
+	// Set SDK version
+	g_ps3_sdk_version = sdk_version;	//hopefully sdk versions betweend main elf and overlay elfs are always the same
```

Comment:
It's probably better to check if the sdk ver of the overlay elf and the main one differ and log it in such case.